### ISSUE TITLE
0.13.0-beta2 multiple problems with terraform_remote_state

### ISSUE THREAD

**Issue by Bobonium** [hashicorp/terraform#25296]
### Terraform Version

```
Terraform v0.13.0-beta2
+ provider registry.terraform.io/-/kubernetes v1.11.3
+ provider registry.terraform.io/-/random v2.2.1
+ provider registry.terraform.io/-/tls v2.1.1
+ provider registry.terraform.io/hashicorp/kubernetes v1.11.1
+ provider registry.terraform.io/hashicorp/random v2.2.1
+ provider registry.terraform.io/hashicorp/tls v2.1.1...
```

### Terraform Configuration Files

```hcl
data "terraform_remote_state" "test" {
  backend = "kubernetes"

  config = {
    load_config_file = true
    secret_suffix              = "test"
    namespace = "tfstates"
  }
}

```

### Expected Behavior
3 Independent Problems I found in the Beta with `terraform_remote_state`:

#### 1. 

running terraform init and importing into a newly configured backend (in this case kubernetes) from a local terraform.tfstate file should work even if the file has an entry for a `terraform_remote_state`

#### 2. 

`terraform_remote_state` is visible in the plan output once and is gone after the next apply

#### 3. 

`terraform_remote_state` should not display sensitive output in plan/apply

#### 1. 

I get the same error as in https://github.com/hashicorp/terraform/issues/25148 which seems to indicate that the problem is not completely resolved as of 0.13.0-beta2. I was able to circumvent this by removing the data entry manually from the JSON state file and importing again

#### 2. 

```
Resource actions are indicated with the following symbols:
 <= read (data resources)

Terraform will perform the following actions:

  # data.terraform_remote_state.test will be read during apply
  # (config refers to values not yet known)
 <= data "terraform_remote_state" "test"  {
```
is displayed in every single plan and apply no matter how often it is applied.

#### 3.

I have marked some outputs in my remote state as `sensitive = true` this is working for the output itself, but remote states making use of `terraform_remote_state` display these sensitive outputs in the plan/apply output.

#### 1. 

 - have a local `terraform.tfstate` file, but no `.terraform` directory
    - this file should have the data for a `terraform_remote_state`
 - configure a new backed (i.e. kubernetes)
 - run terraform init and import the local state with `yes
 - https://github.com/hashicorp/terraform/issues/25148 should happen

#### 2.

 - add a new `terraform_remote_state` to any existing code
 - run terraform apply
 - run terraform plan or apply
 - note that it still displays all the values it reads from the remote state file although they should already be added

#### 3. 

 - add a new `terraform_remote_state` where the source state outputs values with `sensitive = true` 
 - run terraform plan or apply
 - note that `sensitive` values are displayed in clear text

**Comment by jbardin** [comment]
Hi @Bobonium,

Thanks for reporting these. Let's split up the issues here, since they are from different causes. 

I'm still not able to reproduce the error in the first case. If you could add some more detail to the linked issue, we can reopen that and try to investigate. 

The second case is being cause by a flaw in the remote state data source, and I can have a PR for that shortly. 

The 3rd issue about the sensitive values being displayed in the plan is known, and we are using #8076 to track the issue. Once we fix #2 however, the remote state data source behavior should work like it previously did, and be silently read during refresh so it doesn't need to be displayed in the plan.

**Comment by Bobonium** [comment]
Hey @jbardin 
thanks for the quick reply. I tried building a minimalistic example of the error as it's reproducible through all of my repositories, but I wasn't able to build a simple example either. 

I then thought it might be related to the fact that I'm using a ~1 year old fork with the initial kubernetes backend draft. So I tried to reproduce it by creating the state with my fork, exporting the statefile with `terraform state pull` and removing the .terraform directory before running init with the 0.13.0-beta2, but even then it doesn't work in a minimalistic example.

The only thing I can tell you for now is that I can reproduce this every time with my real state files and that I can solve it by removing the datasource entry from the statefile for the `terraform_remote_state`. Also I can't even reproduce this by copy & pasting the datasource entry into a minimalistic example statefile and importing that.

I can update if I should find anything else, but if no one else is reporting this issue it can probably be ignored as it's probably related to me upgrading from the quite old fork in that case.
